tcbuilder/backend/deploy.py

```
@@ -325,46 +293,71 @@ def deploy_tezi_image(tezi_dir, src_sysroot_dir, src_ostree_archive_dir,
+    partition, both checked before the truncated file's partition table is
+    touched. The partition's filesystem is grown to match. On failure,
+    cleanup is attempted - the whole image is removed (delete_on_error) or
+    the file is restored to its original size - and the original error is
+    still raised even if that cleanup itself fails.
     """
+    orig_size = os.path.getsize(raw_img)
     # Round up to a whole sector; a non-sector-multiple size can be rejected at 4Kn.
-    new_size = os.path.getsize(raw_img) + int(added_size_kb) * 1024
+    new_size = orig_size + int(added_size_kb) * 1024
     new_size = (new_size + sector_size - 1) // sector_size * sector_size
     subprocess.check_output(["truncate", "-s", str(new_size), raw_img])
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)